### PR TITLE
feat: Verify genesis every time starting

### PR DESCRIPTION
**Pull request by keroro520**
* Verify genesis block by `BlockVerifier` every time starting node
* Fix capacity calculation and inputs of genesis block
* Verify occupied_capacity of celbase transactions 
* Refactor `tests/uncle_verifier.rs` and add several tests
* **BREAKING CHANGE**: Update genesis hash

### CODE REVIEW

File: spec/src/lib.rs [hunk 1]
```diff
@@ -247,12 +256,6 @@ pub mod test {
         let chain_spec = dev.unwrap();
         let tx = chain_spec.build_system_cell_transaction().unwrap();
 
-        // Tx and Output hash will be used in some test cases directly, assert here for convenience
-        assert_eq!(
-            format!("{:x}", tx.hash()),
-            "48168c5b2460bfa698f60e67f08df5298b1d43b2da7939a219ffd863e1380d11"
-        );
-
```
Comment (keroro520) [inline spec/src/lib.rs]: This test makes no sense.

File: util/app-config/src/lib.rs [hunk 2]
```diff
@@ -217,17 +215,7 @@ fn is_daemon(subcommand_name: &str) -> bool {
...
-        ExitCode::Config
-    })
-}
-
-fn verify_genesis(consensus: Consensus) -> Result<Consensus, Box<Error>> {
-    MerkleRootVerifier::new()
-        .verify(consensus.genesis_block())
-        .map_err(Box::new)?;
-    Ok(consensus)
+    spec.to_consensus().map_err(|_| ExitCode::Config)
```
Comment (keroro520) [inline util/app-config/src/lib.rs]: `spec/src/lib.rs` only ensures that genesis is matched with the specs, recorded in `specs/*.toml`.

`src/subcommand/run.rs` do the real block validation (Unable to do this inside `spec/src/lib.rs` cause circular dependency of `ckb_shared` and `ckb-chain-specs`).

File: verification/src/block_verifier.rs [hunk 3]
```diff
@@ -97,6 +96,16 @@ impl<CP: ChainProvider + Clone> CellbaseVerifier<CP> {
             return Err(Error::Cellbase(CellbaseError::InvalidOutput));
         }
 
+        let invalid_capacity = cellbase_transaction.outputs().iter().any(|op| {
```
Comment (quake) [inline verification/src/block_verifier.rs]: I think that it's better to add a function to `CellOutput`
```
pub fn is_occupied_capacity_overflow(&self) {
    
}
```

And refactoring this code block to:
```
    if cellbase_transaction.outputs().iter().any(CellOutput::is_occupied_capacity_overflow) {
        return Err(Error::CapacityOverflow);
    }
```

Similar code of `CapacityVerifier` can be refactored to use this function also.

File: spec/src/lib.rs (hunk 1 above)
Comment (doitian) [reply spec/src/lib.rs]: This is for developers who need to hard-coded `always_success` hash. However, it is going to be removed in #562 .

File: spec/src/lib.rs [hunk 4]
```diff
@@ -168,19 +169,27 @@ impl ChainSpec {
                 c.get()
                     .map_err(|err| Box::new(err) as Box<Error>)
                     .and_then(|data| {
-                        let data = data.into_owned();
                         // TODO: we should provide a proper lock script here so system cells
                         // can be updated.
-                        Capacity::bytes(data.len())
-                            .map(|cap| CellOutput::new(cap, data, Script::default(), None))
-                            .map_err(|err| Box::new(err) as Box<Error>)
+                        let data = data.into_owned();
```
[the remark this answers is not in the document]
Comment (keroro520) [reply spec/src/lib.rs]: Thanks! It is more clear. Applied and rebased the suggested code.

File: verification/src/block_verifier.rs (hunk 3 above)
Comment (keroro520) [reply verification/src/block_verifier.rs]: Thanks. Updated and rebased.

File: spec/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (yangby-cryptape) [reply spec/src/lib.rs]: IMHO, we shouldn't pass a `capacity` to `CellOutput` as an parameter, `capacity` should be computed in `CellOutput::new()`.
(Not in this PR, we can do it later.)

**Comment by keroro520** [comment]
> Need update `test/integration.toml` genesis hash.

Updated.
And added monitor for child process in `test/` https://github.com/nervosnetwork/ckb/pull/582/files#diff-750b6a3c46becef8048e261f521c1d93

**Pull request merged**